File: pyworks/ch05/math1.py
# 수학 모듈 - math
# 절대값, 올림, 내림, 제곱근, 원주율
import math

# math의 속성
print(math.pi)  # 원주율
print(math.ceil(2.54)) # 정수로 올림
print(math.floor(4.923)) # 내림(버림)
print(math.sqrt(25)) # 제곱근

# 내장 함수
print(round(11.78))

# 절대값
print(abs(-10))

# 절대값을 함수로 만들기
# 1. 알고리즘
"""
def my_abs(x):
    if x < 0:
        return -x
    else:
        return x

print(my_abs(-5))   # -5 -> -(-5), 5 -> 5
print(my_abs(5))
"""

# 2. 알고리즘2
def my_abs2(x):
    y = x * x
    return math.sqrt(y)

# 제곱수(양수로 됨) -> 제곱근(실수로 표기)
print(int(my_abs2(-5)))
print(int(my_abs2(5)))

File: javaworks/ch03/src/methods/Math1.java
package methods;

public class Math1 {

	public static void main(String[] args) {
		// Math 클래스
		//절대값 구하는 함수 사용 - abs()
		//new를 클래스는 클래스 이름으로 직접 접근
		int v1 = Math.abs(-7);
		System.out.println("v1 = " + v1);
		
		//반올림
		long v2 = Math.round(5.7);  
		System.out.println("v2 = " + v2);
		
		//버림
		double v3 = Math.floor(2.12);
		System.out.println("v3 = " + v3);
		
		//랜덤 - 무작위수(0 <= rand < 1)
		double rand = Math.random();
		System.out.println("rand = " + rand);
		
		//가위 바위 보(1, 2, 3)
		int 가위바위보 = (int)(Math.random()*3) + 1;
		System.out.println("가위바위보 = " + 가위바위보);
	}

}
